utils/pattern/src/frontend/mod.rs

```diff
@@ -51,6 +60,17 @@ impl<Backend, Store> Pattern<Backend, Store> {
     pub fn take_store(self) -> Store {
         self.store
     }
+
+    /// # Safety
+    ///
+    /// The store must come from a valid `Pattern` with this `Backend`,
+    /// such as by calling [`Pattern::take_store()`].
+    pub const unsafe fn from_store_unchecked(store: Store) -> Self {
```

Comment:
question/issue: is this a memory safety requirement? if not, this shouldn't be unsafe